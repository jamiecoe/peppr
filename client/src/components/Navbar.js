import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { logoutUser } from '../actions/auth';

export class Navbar extends Component {
  logout = () => {
    this.props.logoutUser();
  }

  render() {
    let className = '';
    switch (this.props.page) {
      case 'singleRecipe':
        className = 'navbar navbar-mobile-hidden';
        break;
      case 'addRecipe':
        className = 'navbar addrecipe__navbar';
        break;
      default:
        className = 'navbar';
    }

    return (
      <nav className={className}>
        <ul>
          <li className="navbar__homelogo">
            <Link to="/recipes">PEPPR</Link>
          </li>
          <li className="navbar__link">
            <Link to="/recipes">
              <i className="ion-ios-nutrition-outline navbar__icon" />
              Recipes
            </Link>
          </li>
          <li className="navbar__link">
            <Link to="/addrecipe">
              <i className="ion-ios-plus-outline navbar__icon" />
              Add
            </Link>
          </li>
          <li className="navbar__link">
            <Link to="/" onClick={this.logout}>
              <i className="ion-ios-undo-outline navbar__icon" />
              Log out
            </Link>
          </li>
        </ul>
      </nav>
    );
  }
}

export default connect(null, { logoutUser })(Navbar);
