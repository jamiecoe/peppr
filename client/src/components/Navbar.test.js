import React from 'react';
import { shallow } from 'enzyme';
import { Navbar } from './Navbar';

describe('Navbar', () => {
  const mockLogout = jest.fn();
  const props = { page: 'singleRecipe', logoutUser: mockLogout };
  const navbar = shallow(<Navbar {...props} />);

  it('renders properly', () => {
    expect(navbar).toMatchSnapshot();
  });

  it('uses props to set the class of <nav> element', () => {
    expect(navbar.find('nav').hasClass('navbar navbar-mobile-hidden'))
      .toBe(true);
  });

  describe('when a user wants to logout', () => {
    beforeEach(() => navbar.find('Link').last().simulate('click'));
    it('calls this.props.logout() when logout Link clicked', () => {
      expect(mockLogout).toHaveBeenCalled();
    });
  });
});
