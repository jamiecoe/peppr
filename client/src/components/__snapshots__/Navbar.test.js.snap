// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar
    logoutUser={[Function]}
    page="singleRecipe"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <li
                className="navbar__homelogo"
        >
                <Link
                        replace={false}
                        to="/recipes"
                >
                        PEPPR
                </Link>
        </li>
        <li
                className="navbar__link"
        >
                <Link
                        replace={false}
                        to="/recipes"
                >
                        <i
                                className="ion-ios-nutrition-outline navbar__icon"
                        />
                        Recipes
                </Link>
        </li>
        <li
                className="navbar__link"
        >
                <Link
                        replace={false}
                        to="/addrecipe"
                >
                        <i
                                className="ion-ios-plus-outline navbar__icon"
                        />
                        Add
                </Link>
        </li>
        <li
                className="navbar__link"
        >
                <Link
                        onClick={[Function]}
                        replace={false}
                        to="/"
                >
                        <i
                                className="ion-ios-undo-outline navbar__icon"
                        />
                        Log out
                </Link>
        </li>
</ul>,
      "className": "navbar navbar-mobile-hidden",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="navbar__homelogo"
>
            <Link
                        replace={false}
                        to="/recipes"
            >
                        PEPPR
            </Link>
</li>,
          <li
            className="navbar__link"
>
            <Link
                        replace={false}
                        to="/recipes"
            >
                        <i
                                    className="ion-ios-nutrition-outline navbar__icon"
                        />
                        Recipes
            </Link>
</li>,
          <li
            className="navbar__link"
>
            <Link
                        replace={false}
                        to="/addrecipe"
            >
                        <i
                                    className="ion-ios-plus-outline navbar__icon"
                        />
                        Add
            </Link>
</li>,
          <li
            className="navbar__link"
>
            <Link
                        onClick={[Function]}
                        replace={false}
                        to="/"
            >
                        <i
                                    className="ion-ios-undo-outline navbar__icon"
                        />
                        Log out
            </Link>
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/recipes"
>
              PEPPR
</Link>,
            "className": "navbar__homelogo",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "PEPPR",
              "replace": false,
              "to": "/recipes",
            },
            "ref": null,
            "rendered": "PEPPR",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/recipes"
>
              <i
                            className="ion-ios-nutrition-outline navbar__icon"
              />
              Recipes
</Link>,
            "className": "navbar__link",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="ion-ios-nutrition-outline navbar__icon"
/>,
                "Recipes",
              ],
              "replace": false,
              "to": "/recipes",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "ion-ios-nutrition-outline navbar__icon",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "Recipes",
            ],
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/addrecipe"
>
              <i
                            className="ion-ios-plus-outline navbar__icon"
              />
              Add
</Link>,
            "className": "navbar__link",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="ion-ios-plus-outline navbar__icon"
/>,
                "Add",
              ],
              "replace": false,
              "to": "/addrecipe",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "ion-ios-plus-outline navbar__icon",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "Add",
            ],
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              onClick={[Function]}
              replace={false}
              to="/"
>
              <i
                            className="ion-ios-undo-outline navbar__icon"
              />
              Log out
</Link>,
            "className": "navbar__link",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <i
                  className="ion-ios-undo-outline navbar__icon"
/>,
                "Log out",
              ],
              "onClick": [Function],
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "ion-ios-undo-outline navbar__icon",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "Log out",
            ],
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li
                    className="navbar__homelogo"
          >
                    <Link
                              replace={false}
                              to="/recipes"
                    >
                              PEPPR
                    </Link>
          </li>
          <li
                    className="navbar__link"
          >
                    <Link
                              replace={false}
                              to="/recipes"
                    >
                              <i
                                        className="ion-ios-nutrition-outline navbar__icon"
                              />
                              Recipes
                    </Link>
          </li>
          <li
                    className="navbar__link"
          >
                    <Link
                              replace={false}
                              to="/addrecipe"
                    >
                              <i
                                        className="ion-ios-plus-outline navbar__icon"
                              />
                              Add
                    </Link>
          </li>
          <li
                    className="navbar__link"
          >
                    <Link
                              onClick={[Function]}
                              replace={false}
                              to="/"
                    >
                              <i
                                        className="ion-ios-undo-outline navbar__icon"
                              />
                              Log out
                    </Link>
          </li>
</ul>,
        "className": "navbar navbar-mobile-hidden",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="navbar__homelogo"
>
              <Link
                            replace={false}
                            to="/recipes"
              >
                            PEPPR
              </Link>
</li>,
            <li
              className="navbar__link"
>
              <Link
                            replace={false}
                            to="/recipes"
              >
                            <i
                                          className="ion-ios-nutrition-outline navbar__icon"
                            />
                            Recipes
              </Link>
</li>,
            <li
              className="navbar__link"
>
              <Link
                            replace={false}
                            to="/addrecipe"
              >
                            <i
                                          className="ion-ios-plus-outline navbar__icon"
                            />
                            Add
              </Link>
</li>,
            <li
              className="navbar__link"
>
              <Link
                            onClick={[Function]}
                            replace={false}
                            to="/"
              >
                            <i
                                          className="ion-ios-undo-outline navbar__icon"
                            />
                            Log out
              </Link>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/recipes"
>
                PEPPR
</Link>,
              "className": "navbar__homelogo",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "PEPPR",
                "replace": false,
                "to": "/recipes",
              },
              "ref": null,
              "rendered": "PEPPR",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/recipes"
>
                <i
                                className="ion-ios-nutrition-outline navbar__icon"
                />
                Recipes
</Link>,
              "className": "navbar__link",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <i
                    className="ion-ios-nutrition-outline navbar__icon"
/>,
                  "Recipes",
                ],
                "replace": false,
                "to": "/recipes",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "ion-ios-nutrition-outline navbar__icon",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "Recipes",
              ],
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/addrecipe"
>
                <i
                                className="ion-ios-plus-outline navbar__icon"
                />
                Add
</Link>,
              "className": "navbar__link",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <i
                    className="ion-ios-plus-outline navbar__icon"
/>,
                  "Add",
                ],
                "replace": false,
                "to": "/addrecipe",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "ion-ios-plus-outline navbar__icon",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "Add",
              ],
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                onClick={[Function]}
                replace={false}
                to="/"
>
                <i
                                className="ion-ios-undo-outline navbar__icon"
                />
                Log out
</Link>,
              "className": "navbar__link",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <i
                    className="ion-ios-undo-outline navbar__icon"
/>,
                  "Log out",
                ],
                "onClick": [Function],
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "ion-ios-undo-outline navbar__icon",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "i",
                },
                "Log out",
              ],
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
